tests/farmer_harvester/test_farmer.py

```diff
@@ -619,6 +638,274 @@ def assert_pool_errors_24h() -> None:
+        pool_state[key] = value
+    return pool_state
+
+
+@pytest.mark.parametrize(
+    argnames="pool_response_case,expected_pool_state",
+    argvalues=[
+        pytest.param(
+            DummyPoolResponse(True, 200, new_difficulty=123),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "points_acknowledged_since_start": 123,
+                    "points_acknowledged_24h": [123],
+                    "valid_partials_since_start": 1,
+                    "valid_partials_24h": [1],
+                }
+            ),
+            id="valid_response",
+        ),
+        pytest.param(
+            DummyPoolResponse(False, 500),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "invalid_partials_since_start": 1,
+                    "invalid_partials_24h": [1],
+                }
+            ),
+            id="response_not_ok",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.TOO_LATE.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.TOO_LATE.value)}],
+                    "stale_partials_since_start": 1,
+                    "stale_partials_24h": [1],
+                }
+            ),
+            id="stale_partial",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.PROOF_NOT_GOOD_ENOUGH.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.PROOF_NOT_GOOD_ENOUGH.value)}],
+                    "insufficient_partials_since_start": 1,
+                    "insufficient_partials_24h": [1],
+                }
+            ),
+            id="insufficient_partial",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.SERVER_EXCEPTION.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.SERVER_EXCEPTION.value)}],
+                    "invalid_partials_since_start": 1,
+                    "invalid_partials_24h": [1],
+                }
+            ),
+            id="other_failed_partial",
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_farmer_pool_response(
+    mocker: MockerFixture,
+    harvester_farmer_environment: HarvesterFarmerEnvironment,
```

Comment:
We're only using farmer API in this test setup, so maybe we should use a lighter fixture such as `farmer_one_harvester`?

Reply:
Done